### ISSUE TITLE
No Last.fm scrobbling at all

### ISSUE THREAD

**Issue by ronilaukkarinen** [airsonic/airsonic#785]
## Problem description

When playing songs, I don't see any scrobbles on my Last.fm page.

### Steps to reproduce

Set up your Last.fm settings, username and password, see no scrobbles.

## System information

 * **Airsonic version**: *v10.1.2*
 * **Operating system**: *Fedora 25*
 * **Proxy server**: None
 * **Client**: MacOS Google Chrome Version 68.0.3440.106 (Official Build) (64-bit)

## Additional notes

Tried to set up username and password many times with no luck. No log items either.

**Comment by jooola** [comment]
Maybe increase logging level ?

**Comment by jooola** [comment]
https://airsonic.github.io/docs/logging/

**Comment by ronilaukkarinen** [comment]
Changed all logging to DEBUG. When I restart airsonic I get flood to the log, but when saving credentials, no entries in the log. No entries either while playing songs, with `tail -f /var/airsonic/airsonic.log`.

**Comment by fxthomas** [comment]
Can you try again with the last version?

**Comment by ronilaukkarinen** [comment]
@fxthomas Sorry for the late reply, been busy. Just got newest master and built from source. Ticked off Last.fm and saved, then ticked it back on, added my credentials and saved. Then shuffled everything and started to play, getting:

```
2019-12-12 11:44:45.574  WARN --- o.a.p.s.s.LastFMScrobbler                : Failed to scrobble song 'Luciferin valo' at Last.fm.  Unknown response: <html><head>
```

Sadly, still no scrobbles :(

**Comment by fxthomas** [comment]
Hum hum. Looks like it's getting some HTML back, not proper API results?

I don't use Last.fm, but there is a bunch of issues about Airsonic using a decades-old version of the Last.fm API (#776), maybe this is linked to your issue?

**Comment by ronilaukkarinen** [comment]
Yeah, probably. Can't find any endpoints from source, but then again I'm not a Java expert. Current api works via url simply by calling https://ws.audioscrobbler.com/2.0/?method=user.getrecenttracks&user=rolle-&api_key=YOUR_API_KEY&limit=200&page=1500&format=json and it seems the current airsonic uses 1.0 which is obsolete... :/

**Comment by eharris** [comment]
Yes, #776 was an issue asking to update the last.fm scrobbling API, since the old one being used is now obsolete.  But as far as I know no one has been interested enough to submit a PR to update it (I don't use last.fm myself).  Since #1224 has now been integrated, the scrobbling code has been reorganized a bit better.

Should this be closed in favor of #776?

**Comment by ronilaukkarinen** [comment]
Isn't #1224 for ListenBrainz only? As far as I know, Last.fm still does not work, just tested the latest master:

```
2020-01-16 10:19:07.452  WARN --- o.a.p.s.s.LastFMScrobbler                : Failed to scrobble song 'Death' at Last.fm.  Unknown response: <html><head>
```

**Comment by fxthomas** [comment]
Yup, #1224 is for ListenBrainz only (and works, at least on my instance). Last.fm support was not touched...

**Comment by SyntrophicConsortium** [comment]
I'm running a newer version of Airsonic than what this issue pertains to, but it doesn't feel appropriate to open a new issue.

Last.fm used to work for me, but it stopped working as soon as I updated to Airsonic 10.6.0 (a couple days ago). I'm seeing errors in the Airsonic log like this: 

> 2020-04-18 22:31:37.408 WARN --- o.a.p.s.s.LastFMScrobbler : Failed to scrobble song 'we used to know' at Last.fm. Unknown response: <html><head>

Last.fm has not mentioned any issues on their twitter, aside from a recent problem with their Spotify scrobbling service specifically. 

I know nothing about Listenbrainz, but I guess I can look into that. As long as I can import my 15-year-old last.fm play history, I don't mind moving over to that eventually.

**Comment by randomnicode** [comment]
@SyntrophicConsortium 
This is a bug in airsonic code that was pointed out here: https://github.com/airsonic/airsonic/issues/776#issuecomment-593212103 along with its fix, but it seems no one paid attention. Warning that master cannot scrobble was provided well in advance of the release. It was fixed in [`airsonic-advanced`](https://github.com/airsonic-advanced/airsonic-advanced) (the PR is linked in the comment). @fxthomas It's a one-character-change. There are several other bugs like this littered across the codebase that are latent and haven't been discovered yet.

**Comment by fxthomas** [comment]
Ah, thank you for keeping up with this! I do not follow Last.fm closely (including the discussion your comment was in), and my understanding was that Last.fm discontinued the old v1 API and that it stopped working entirely.

Will probably try to get this included in yet another bugfix release this week-end.

**Comment by fxthomas** [comment]
Can you guys try https://github.com/fxthomas/airsonic/releases/tag/fix-lastfm-10.6 and let me know if it works for you?

**Comment by Pollinopops** [comment]
Works again for me ! Thanks for the quick fix !
